master/internal/api_user.go

```diff
@@ -158,17 +149,20 @@ func (a *apiServer) PatchUser(
 	if !curUser.Admin && curUser.Username != req.Username {
 		return nil, grpcutil.ErrPermissionDenied
 	}
-	user := &model.User{Username: req.Username}
 	// TODO: handle any field name:
+	updateUser := &model.User{Username: req.Username}
 	if req.User.DisplayName != "" {
-		user.DisplayName = null.StringFrom(req.User.DisplayName)
-		switch err = a.m.db.UpdateUser(user, []string{"display_name"}, nil); {
-		case err == db.ErrNotFound:
-			return nil, errUserNotFound
-		case err != nil:
+		updateUser.DisplayName = null.StringFrom(req.User.DisplayName)
+	}
+	if req.User.Password != nil {
+		if err = updateUser.UpdatePasswordHash(req.User.Password.Value); err != nil {
 			return nil, err
 		}
 	}
-	fullUser, err := getUser(a.m.db, req.Username)
-	return &apiv1.PatchUserResponse{User: fullUser}, err
+	// return the finalUser object so we can be assured only public fields are passed on
+	finalUser := &userv1.User{}
+	err = a.m.db.QueryProto(
+		"update_user", finalUser, updateUser.Username, updateUser.DisplayName, updateUser.PasswordHash)
```

Comment:
if this CLI is going to use this, doesn't either this or the CLI need to add the client side hash? if the webui is still sending it hashed, i think this would break.